### PR TITLE
show closest

### PR DESCRIPTION
**Pull request by lieblichaaron**
## ℹ️ Change description:
Nothing here

<details><summary><sub>templates<sub></summary>
<p>
  
<sub>[templates: [bug](?expand=1&template=bug.md), [feature](?expand=1&template=feature.md)]</sub>

</p>
</details>

### CODE REVIEW

**Review by avivbarniv — APPROVED** [review]
You can use https://turfjs.org/docs/#explode

File: examples/client/Locomotion/src/pages/ActiveRide/newMap.js [hunk 1]
```diff
@@ -129,6 +131,56 @@ export default React.forwardRef(({
...
+    let coordsToFindClosestTerritory;
+    const useStopPoints = requestStopPoints.filter((sp => sp.lat)).length > 1;
+    if (useStopPoints) {
+      const [pickup] = requestStopPoints;
+      coordsToFindClosestTerritory = {
+        latitude: pickup.lat,
+        longitude: pickup.lng,
+      };
+    } else {
+      let location = await getPosition();
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/newMap.js]: remove, we want only the stop point

File: examples/client/Locomotion/src/pages/ActiveRide/newMap.js [hunk 2]
```diff
@@ -129,6 +131,56 @@ export default React.forwardRef(({
...
+      if (!location) {
+        location = DEFAULT_COORDS;
+      }
+      // eslint-disable-next-line prefer-destructuring
+      coordsToFindClosestTerritory = location.coords;
+    }
+
+    const allTerritoryPoints = territory.map(({ polygon }) => polygon.coordinates).flat().flat()
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/newMap.js]: const allTerritoryPoints = explode(territory.map(({ polygon }) => polygon.coordinates).flat().flat())
    
    I think so

File: examples/client/Locomotion/src/pages/ActiveRide/newMap.js [hunk 3]
```diff
@@ -129,6 +131,56 @@ export default React.forwardRef(({
...
+    const targetPoint = point(
+      [
+        parseFloat(coordsToFindClosestTerritory.latitude),
+        parseFloat(coordsToFindClosestTerritory.longitude),
+      ],
+    );
+    const points = featureCollection(allTerritoryPoints);
+    const nearest = nearestPoint(targetPoint, points);
+
+    const closestTerritory = territory.find(bm => bm.polygon.coordinates[0].find(coord => coord[0] === nearest.geometry.coordinates[1]
```
Comment (avivbarniv) [inline examples/client/Locomotion/src/pages/ActiveRide/newMap.js]: you can use https://turfjs.org/docs/#booleanPointInPolygon if you want to find the territory

Comment (lieblichaaron) [reply examples/client/Locomotion/src/pages/ActiveRide/newMap.js:163]: im not trying to find in polygon. if its outside it wont find it this way

**Pull request merged**